tools/ds-identify

```diff
@@ -281,6 +280,50 @@ read_fs_info() {
     DI_ISO9660_DEVS="${isodevs#,}"
 }
 
+read_geom() {
+    local oifs="$IFS" line="" delim=","
+    local ret=0 out="" labels="" dev="" label="" ftype="" isodevs=""
+    out=$(geom label status -as) || {
+        ret=$?
+        error "failed running [$ret]: geom label status -as"
+        DI_FS_LABELS="$UNAVAILABLE:error"
+        DI_ISO9660_DEVS="$UNAVAILABLE:error"
+        return $ret
+    }
+
+    # shellcheck disable=2086
+    { IFS="$CR"; set -- $out; IFS="$oifs"; }
+
+    for line in "$@"; do
+        # shellcheck disable=2086
+        set -- $line
+        provider=$1
+        ftype="${provider%/*}"
+        label="${provider#*/}"
+        dev=$3
+
+        [ -n "$dev" -a "$ftype" = "iso9660" ] &&
+            isodevs="${isodevs},${dev}=$label"
+
+        labels="${labels}${label}${delim}"
+    done
+
+    DI_FS_LABELS="${labels%${delim}}"
+    DI_ISO9660_DEVS="${isodevs#,}"
+}
+
+read_fs_info() {
+    cached "${DI_BLKID_OUTPUT}" && return 0
```

Comment:
bah. this needs fixing. we dont actually do anything to set that variable.
the intent of the variable is to say we've already run blkid and cached its output.

this isn't your fault. the intent of DI_BLKID_OUTPUT was to store the result of calling 'blkid', so that we didn't call that external program more than once.  So I guess what I'd like from you:

a.) a comment in read_fs_info stating what variables it sets (DI_FS_LABELS, DI_ISO9660_DEVS, DI_FS_UUIDS)
b.) drop of the DI_BLKID_OUTPUT cache here
c.) add caching of call to 'geom'. the easiest way to do this might be just adding a variable 'DI_GEOM' and put the call to 'geom' in a function, something like:

    geom_label_status_as() {
        # call the bsd program geom '-as', set DI_GEOM_LABEL_STATUS_OUT
        cached "$DI_GEOM_LABEL_STATUS_OUT" && return 0
        local out=""
        out=$(geom -as label status) &&  DI_GEOM_LABEL_STATUS="$out" ||
            DI_GEOM_LABEL_STATUS="$UNAVAILABLE"
        return
     }
         

d.) fixing of the caching of 'blkid' output (i realize i'm asking you to fix a bug that isn't your fault... go ahead and git blame this and see that it is mine. :wink: ), but having this variable/cache that doesn't do anything is just confusing.

Reply:
> i realize i'm asking you to fix a bug that isn't your fault...

i'm perfectly happy to do that, it just means this patch will take as long as all my other patches usually do 😅

Reply:
i *think* this is done now